### PR TITLE
Add pull requests guide doc

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

**Review by jeffschwMSFT — APPROVED** [review]
lgtm

**Review by jkotas — APPROVED** [review]

File: CONTRIBUTING.md
Comment (CoffeeFlux) [inline CONTRIBUTING.md]:
```suggestion
4. Make and commit your changes to your branch.
```

File: CONTRIBUTING.md
Comment (CoffeeFlux) [inline CONTRIBUTING.md]:
```suggestion
8. Wait for feedback or approval of your changes from the area owners.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
To merge pull requests, you must have write permissions in the repository. If you are a member of the .NET team or have an official partnership you can request access [here](https://repos.opensource.microsoft.com/dotnet/teams/dotnet-corefx/join/).
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
* Do not mix unrelated changes in one pull request. For example, a code style change should never be mixed with a bug fix.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
* All changes should follow the existing code style. You can read more about different code styles at [docs/coding-guidelines](coding-guidelines/).
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
* Use Draft pull requests for changes you are still working on but want early CI loop feedback. When you think your changes are ready for review, [change the status](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request) of your pull request.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
If during the code review process a merge conflict occurs the area owner is responsible for its resolution. Pull requests should not be on hold due to the author's unwillingness to resolve code conflicts. GitHub makes this easier by allowing simple conflict resolution using the [conflict-editor](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github).
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
 * You can request another review from the original reviewer.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
 * You can trigger a rebuild by commenting `/arp run` or manually re-run only the failing lanes in Azure DevOps tab.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
 * Reach out to the infrastructure team for assistance on Teams [channel](https://teams.microsoft.com/l/channel/19%3ab27b36ecd10a46398da76b02f0411de7%40thread.skype/Infrastructure?groupId=014ca51d-be57-47fa-9628-a15efcc3c376&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) in other cases.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
Please always squash the pull request unless explicitly requested by the PR author, even if the PR contains only one commit. It creates a simpler history than a Merge Commit.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [inline docs/pr-guide.md]:
```suggestion
If for whatever reason you would like to move your pull request back to an in-progress status to avoid merging it in the current form, you can do that by adding [WIP] prefix to the pull request title.
```

File: docs/pr-guide.md
Comment (CoffeeFlux) [reply docs/pr-guide.md]: [This](https://github.com/dotnet/runtime/blob/master/docs/issues-pr-management.md#is-this-a-squash-on-merge-repo) should probably also be updated alongside this comment, since it's currently unclear (presumably the snippet was copy-pasted and made sense in context wherever it came from).

File: CONTRIBUTING.md
Comment (danmosemsft) [inline CONTRIBUTING.md]:
```suggestion
3. In your fork, create a branch off of master (`git checkout -b mybranch`).
```

File: CONTRIBUTING.md
Comment (danmosemsft) [inline CONTRIBUTING.md]:
```suggestion
8. Wait for feedback or approval of your changes from the [area owners](https://github.com/dotnet/runtime/blob/master/docs/area-owners.md).
```

File: CONTRIBUTING.md
Comment (danmosemsft) [inline CONTRIBUTING.md]:
```suggestion
9. When Microsoft employee(s) have signed off, and all checks are green, your change will be merged.
```

File: docs/pr-guide.md
Comment (danmosemsft) [reply docs/pr-guide.md]: @safern is `/arp run` what we recommend?

File: docs/pr-guide.md
Comment (danmosemsft) [reply docs/pr-guide.md]: This Teams channel is only accessible from Microsoft corpnet, I think?

File: docs/pr-guide.md
Comment (danmosemsft) [inline docs/pr-guide.md]:
```suggestion
Please always squash the pull request unless there are special circumstances. Do so, even if the PR contains only one commit. It creates a simpler history than a Merge Commit. "Special circumstances" are rare, and typically mean that there are a series of cleanly separated changes that will be too hard to understand if squashed together, or for some reason we want to preserve the ability to bisect them.
```

Ie., - request of the PR author is _not_ sufficient reason to not squash. (Some contributors prefer their changes not be squashed, but this is not repo policy.)

**Review by danmoseley — APPROVED** [review]

File: CONTRIBUTING.md
Comment (marek-safar) [reply CONTRIBUTING.md]: Did you mean committer(s) or reviewer(s)?

File: docs/pr-guide.md
Comment (safern) [reply docs/pr-guide.md]: Actually it is `/azp run <pipeline name>` (i.e `/azp run runtime`). In fact, we don't recommend `/azp run` as it triggers ALL the outerloop and non-required builds and we're working with AzDo on providing a feature were we can disable that behavior.

File: docs/pr-guide.md
Comment (safern) [inline docs/pr-guide.md]: This should be `/azp run <pipeline-name>`. 

See my comment:
> Actually it is /azp run <pipeline name> (i.e /azp run runtime). In fact, we don't recommend /azp run as it triggers ALL the outerloop and non-required builds and we're working with AzDo on providing a feature were we can disable that behavior.

**Review by safern — APPROVED** [review]

File: docs/pr-guide.md
Comment (marek-safar) [reply docs/pr-guide.md]: I don't think that's helpful unless we explain somewhere what <pipeline-name> is.

Comment (safern) [reply docs/pr-guide.md]: Why it wouldn't be helpful saying people shouldn't use `/azp run`? And what I mean, is that `arp run` is not correct it is `/azp run`. 

```suggestion
 * You can trigger a rebuild by commenting `/azp run <pipeline name>` or manually re-run only the failing lanes in Azure DevOps tab.
```

File: CONTRIBUTING.md
Comment (danmosemsft) [reply CONTRIBUTING.md]: Community reviews are welcome but I do not think we will merge something unless someone with commit rights has reviewed it.

Almost all area owners have commit rights. But we do not require an area owner to sign off to the change -- it is based on best judgement about the change.

I"ll let you pick the words though 👍

File: docs/pr-guide.md
Comment (danmosemsft) [reply docs/pr-guide.md]: @safern I think he's asking how to get a list of pipelines. Eg., /azp help or something,

File: docs/pr-guide.md
Comment (danmosemsft) [inline docs/pr-guide.md]: Might have got lost in another comment. This is only accessible to folks authenticated into Microsoft corpnet. I don't think it matters if we put the link here, but it's probably worth a note explaining that community members can't access it. In the future, it might be interesting to have a channel that we all can access.

**Review by CoffeeFlux — APPROVED** [review]

File: docs/pr-guide.md
Comment (CoffeeFlux) [reply docs/pr-guide.md]: Maybe say community members can ask on Gitter, since that's open to everyone?

Comment (danmosemsft) [reply docs/pr-guide.md]: Yeah, that may be better

File: docs/pr-guide.md
Comment (safern) [reply docs/pr-guide.md]: I see, yeah, then we can mention people can get the list of pipelines with `/azp list`

**Pull request merged**
